lib/DiscoveryStreamFeed.jsm

```diff
@@ -357,47 +356,43 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
 
   async loadLayout(sendUpdate, isStartup) {
     let layoutResp = {};
+    let url = "";
     if (!this.config.hardcoded_layout) {
       layoutResp = await this.fetchLayout(isStartup);
     }
 
     if (!layoutResp || !layoutResp.layout) {
-      if (
+      // Set a hardcoded layout if one is needed.
+      // Changing values in this layout in memory object is unnecessary.
+      layoutResp = getHardcodedLayout(
         this.config.hardcoded_basic_layout ||
-        this.store.getState().Prefs.values[PREF_HARDCODED_BASIC_LAYOUT]
-      ) {
-        layoutResp = { lastUpdate: Date.now(), ...basicLayoutResp };
-      } else {
-        layoutResp = { lastUpdate: Date.now(), ...defaultLayoutResp };
-      }
-    }
-
-    if (
-      layoutResp.spocs &&
-      (this.store.getState().Prefs.values[PREF_SPOCS_ENDPOINT] ||
-        this.config.spocs_endpoint)
-    ) {
-      layoutResp.spocs.url =
-        this.store.getState().Prefs.values[PREF_SPOCS_ENDPOINT] ||
-        this.config.spocs_endpoint;
+          this.store.getState().Prefs.values[PREF_HARDCODED_BASIC_LAYOUT]
+      );
     }
 
     sendUpdate({
       type: at.DISCOVERY_STREAM_LAYOUT_UPDATE,
       data: layoutResp,
```

Comment:
The reducer in this dispatch actually ends up just ignoring the value in layoutResp.spocs, so changing it directly was also pointless. So I ended up optimizing with a function that returns a new object each time, but also, not modifying the original value.